server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/FileEntryNode.java

```diff
@@ -26,9 +34,25 @@
 public class FileEntryNode extends Node {
 
   private final String fileName;
+  private final String fileControlClause;
 
-  public FileEntryNode(Locality location, String name) {
+  public FileEntryNode(Locality location, String name, String fileControlClause) {
     super(location, NodeType.FILE_CONTROL_ENTRY);
     this.fileName = name;
+    this.fileControlClause = fileControlClause;
+    addProcessStep(this::processNode);
   }
+
+  private List<SyntaxError> processNode() {
+    this.getNearestParentByType(ROOT)
+        .ifPresent(o -> o.getDepthFirstStream()
+            .filter(n -> n.getNodeType() == NodeType.VARIABLE_DEFINITION)
+            .filter(n -> n instanceof VariableDefinitionNode)
```

Comment:
You can use only:
```
.filter(hasType(NodeType.VARIABLE_DEFINITION))
```
If the node has type `VARIABLE_DEFINITION` it will be an instance of `VariableDefinitionNode` you don't need to filter this again.

Reply:
Fixed